### DISCUSSION TITLE
PostUpgradeTask does not have environment variables

### DISCUSSION THREAD

**Discussion by uluzox** [renovatebot/renovate#28909] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

Hi,
I am trying to use an renovate bot's postUgradeTask to run a task after a dependency upgrade.
The script I am running depends on environment variables of GitLab CI but these are not present with the task runner(?)

```json
  "postUpgradeTasks": {
    "commands": [
      "env",
      "./.husky/pre-commit || true"
    ],
    "fileFilters": [
      "*/**"
    ],
    "executionMode": "branch"
  }
```

How can I make the environment variables of the runner available to the post upgrade task?

### Logs (if relevant)

<details><summary>Logs</summary>

```
DEBUG: Executed post-upgrade task (repository=myrepo, branch=renovate/mydep)
       "dep": "mydep",
       "cmd": "env",
       "stderr": "",
       "stdout": "HOME=/home/ubuntu\nCONTAINERBASE_CACHE_DIR=/renovate-bot/renovate/cache/containerbase\nPATH=/home/ubuntu/.cargo/bin:/home/ubuntu/.local/bin:/go/bin:/home/ubuntu/bin:/home/ubuntu/.npm-global/bin:/home/ubuntu/.cargo/bin:/home/ubuntu/.local/bin:/go/bin:/home/ubuntu/bin:/home/ubuntu/.npm-global/bin:/home/ubuntu/.cargo/bin:/home/ubuntu/.local/bin:/go/bin:/home/ubuntu/bin:/home/ubuntu/.npm-global/bin:/home/ubuntu/.cargo/bin:/home/ubuntu/.local/bin:/go/bin:/home/ubuntu/bin:/home/ubuntu/.npm-global/bin:/home/ubuntu/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin\nLANG=C.UTF-8\nLC_ALL=C.UTF-8\nPWD=/builds/renovate-bot/renovate/repos/gitlab/myrepo\n"
```

</details>

**Comment by rarkins** [comment, marked as answer]
Renovate does not pass through all env to child processes by default. Look at exposeAllEnv or customEnvVariablea

**Reply by uluzox** [reply]
Thank you! Worked like a charm!
